### ISSUE TITLE
Implement an oggreader, and update play-from-disk to use it.

### ISSUE THREAD

**Issue by Esser420** [pion/webrtc#1133]
## Summary

Have an example of how to stream just audio (either from file or microphone)

## Motivation

There are already examples on how to stream purely video, so it would be nice to have the same for audio. I'm pretty sure this is possible already and is therefore not a new feature per-se.

The example holds value for applications that get their video stream separately from their audio stream.

Basically to enable something similar to `getUserMedia()` that is used in the browser.

**Comment by Sean-Der** [comment]
Hey @Esser420, sorry for the delay!

This is a good idea, what we need is an `oggreader` so we can update [play-from-disk](https://github.com/pion/webrtc/tree/master/examples/play-from-disk) to play audio to the user.

This package itself will never support microphones, that requires media pipeline integration (which is out of the scope of the project). But another project [mediadevices](https://github.com/pion/mediadevices) does provide this! You can capture different sources, and then send them via Pion WebRTC.

**Comment by somersbmatthews** [comment]
I am working on implementing oggreader.

**Comment by Sean-Der** [comment]
That is awesome @somersbmatthews !

If you have questions I am in [Slack](https://pion.ly/slack) and happy to answer them :) also feel free to push WIP (if you have tests), doesn't have to be perfect first try.

**Comment by funvit** [comment]
Hi, i was in need of ogg reader, so i wrote one..

Anyone is free to improve this code.


[oggreader.go.zip](https://github.com/pion/webrtc/files/4637231/oggreader.go.zip)

**Comment by Sean-Der** [comment]
Fantastic work @funvit!

Would you be interested in opening a PR? If not I can take your code and write tests/reformat as needed. I would put you as the co-author on the commit.

**Comment by funvit** [comment]
@Sean-Der out of time currently...

**Comment by Sean-Der** [comment]
@somersbmatthews how is your work going? I would love to have this merged under your name, are you able to take any of funvit's code? If we add some tests/minor modifications I think we can land it!

I can do the work myself, but would much rather help get people involved in the project.

**Comment by somersbmatthews** [comment]
@Sean-Der still working on it. I have more code than funvit, a lot of it might be unnecessary. I will slack you when I push by Sunday 5pm.

**Comment by Sean-Der** [comment]
Fantastic! No rush, if you need more time please take it :)

**Comment by somersbmatthews** [comment]
Unlike the ivf filed, the opus files have some headers of variable length, therefore the sound data starts at variable offset in the .opus file. I'm getting it figured out slowly but surely. Also there is some semantic issues with the usage of ogg. As .ogg is Vorbis and .opus is Opus and both are ogg. I will make PR with my suggested usage of these terms.

**Comment by Sean-Der** [comment]
Hey @somersbmatthews! 

Feel free to start with an `oggreader` that only works with Pion's `oggwriter`. I would rather see you land something in master today and improve later.

I see a lot of developers stress about making it perfect, don't worry about that here :)

**Comment by somersbmatthews** [comment]
Does this command work on anyone's machine? `go get github.com/pion/webrtc/v2/examples/play-from-disk`

**Comment by somersbmatthews** [comment]
That command is from here: https://github.com/pion/webrtc/blob/master/examples/play-from-disk/README.md

**Comment by somersbmatthews** [comment]
I don't see how the `play-from-disk` as a command is supposed to work either. I can rewrite the examples like this into something that does work, unless I'm making a mistake somehow.

**Comment by somersbmatthews** [comment]
And cgo is not an option for usage in pion/webrtc right?

**Comment by Sean-Der** [comment]
@somersbmatthews That works for me! You can also try cloning and running `go run *.go`. 

Also `cgo` isn't an option, I don't think we will need one for a minimal `oggreader`! The code shared above really should cover everything we need, I think it really just needs more testing. If you can get something that covers the same API surface we should be good.

Feel free to open a PR even if the functions are stubs, would love to help you start on this :)

**Comment by Sean-Der** [comment]
Hey @somersbmatthews are you able to open a PR soon?

This has become a blocking issue for me. I am working on insertable streams, and need audio from disk support.

If you get a chance I would open a PR with the work you have so far. It doesn't have to be perfect, but would love to get this in.

**Comment by somersbmatthews** [comment]
@Sean-Der ok. I'm having difficulty getting save to disk to work so I can't see if my reader is working. I will push what I have before I go to sleep tonight. It might not be useable tho.
